autoparallel/_passes/graph_multiplex.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
```

Comment:
The only real changes in this PR are `example_llama3_di_dw.py` and `split_di_dw_graph.py`, the rest is an artifact of me working on top of https://github.com/meta-pytorch/autoparallel/pull/205 and making the PR against main. I can move this graph pass somewhere else and/or wait for that other PR to land as necessary